spec/lib/elasticsearch_client_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+require 'vcr_helper'
+
+require 'rex_releases'
+
+RSpec.describe ElasticsearchClient, vcr: VCR_OPTS do
+
+  before {
+    # Even in VCR playback, the code freaks out if these aren't set (in
+    # playback, values don't matter)
+    ENV['AWS_ACCESS_KEY_ID'] ||= 'foo'
+    ENV['AWS_SECRET_ACCESS_KEY'] ||= 'bar'
+  }
+
+  let(:fake_es_domain_name) { "spec-esdomain-#{SecureRandom.hex(7)}" }
+
+  it 'can access a restricted AWS ES domain using signed requests' do
+    TempAwsEnv.make do |env|
+      domain_status = env.create_elasticsearch_domain(name: fake_es_domain_name)
+
+      signing_client = ElasticsearchClient.new(
+        url: "https://#{domain_status.endpoint}",
+        sign_aws_requests: true
+      )
+
+      resp = signing_client.search q: "test"
+      expect(resp).to have_key("took")
+
+      non_signing_client = ElasticsearchClient.new(
```

Comment:
this might be clearer specified as a named subject.

Reply:
I tend to find uses of `subject` (the implicit and unnamed kinds, which I realize is not what you’re talking about) confusing, and as such normally avoid it for all of its variants.  I like this [SO answer](https://stackoverflow.com/a/38459039) I just came across, as your comment made me think about this! I think instead of a named subject I’d just use a `let`, and then we wouldn’t have uses of the anonymous subject which could be awkward when there are potentially multiple subjects as there would be here. 

There are some times I have used an anonymous subject, like when I have many little tests to write about something and it is clear in the spec what the subject is.

Reply:
agree on the use of an implicit subject.  I personally like named subjects (they're pretty much the same thing as a let, but with a tiny bit of extra meaning in that it makes it obvious what your subject is for the test).  this seems like a minor point tho.  Not a big deal imo.  I just tend to like the code to be as obvious as possible...and a bit less rails/ruby magic as possible.  

Going off on a tangent here....Sandi Metz gave us a good point when we met her:  dont be afraid of un-DRYness in your tests, and to lean more toward being descriptive and explicit (over clear and DRY).  DRYness for concepts in the code, but not DRYness just for the sake of dryness. (This was a great meeting with her.  Would be cool of OpenStax can get her in for a week somewhere for all devs!)